### PR TITLE
Updating Readme in Code Reviews

### PR DESCRIPTION
**Pull request by saroup**
With this change I made modification to the readme file in the code-review section and udated the pull request template 
- I updated section about the goals of CRs
- Added more details on healthy code review process.
- refactored some of the other sections.

TODO: There are some sections that still need further details like how to chose PR size, and how to inluce the right SMEs.

#102 #106 #104

### CODE REVIEW

**Comment by fnocera** [comment]
Looks good, thanks for the contribution!

File: code-reviews/pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- Besides writing tests with your change, you should state the manual validation that you have done
+  * [ ] Ran it on specific OS
+  * [ ] Ran it using specific test set
+  * [ ] Description of test scenarios that you have tried.
+
+### Any relevant logs or outputs?
+
+- If you are printing something show a screenshot 
+- When you want to share long logs upload to:
+<br>(StorageAccount)/pr-support/attachments/(PR Number)/(yourFiles) using [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/) or [portal.azure.com](portal.azure.com) and insert the link here.
```
Comment (TessFerrandez) [inline code-reviews/pull-request-template.md]: I would like to suggest a pass through this document with a markdown linter - like https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint for VS Code.

As an example this contains an open HTML br element that doesn't render well in all readers.  
It also starts with a ### 3rd level title, and a few more things like that.  

This comment applies to the readme.md as well.

File: code-reviews/pull-request-template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+### Other information or known dependencies
+- Any other information or known dependencies that is important to this PR.
+- ToDos that are to be done after this PR.
```
Comment (TessFerrandez) [inline code-reviews/pull-request-template.md]: Nit:  I like the way you wrote your TODO with capital letters in the PR for this, should this also be capital letters.

File: code-reviews/readme.md [hunk 3]
```diff
@@ -1,140 +1,160 @@
...
+* [ ] All new and existing tests pass.
+* [ ] The code your wrote follows the style of the project.
+* [ ] Run linting checks.
+* [ ] Do manual testing (run the code) and make sure to not introduce a breaking change.
+3. Once done with a unit of code submit a pull request (PR) that follows [pull-request-template.md](pull-request-template.md)
+4. Add one or two reviewers (depending on your project's guidelines). Ideally, you would add at least someone who has expertise and is familiar with the project or the language used and someone who is less familiar with the project or language.
+5. Be open to receive feedback, discuss design/code logic and address all comments as follows:
+   - Resolve the comment if you will make the requested change
+   - Mark the comment as won't fix if you are not going to make the requested changes and explain clearly the reasoning
+   - Ask clarification questions on the review itself(not in a private chat) if you don't understand a reviewer's comment
```
Comment (TessFerrandez) [inline code-reviews/readme.md]: Nit: Space before (not in a private chat)

File: code-reviews/readme.md [hunk 4]
```diff
@@ -1,140 +1,160 @@
...
+    1.  [ ] If the code involves a user-facing change, is there a GIF/photo that explains the functionality? If not, it might be key to validate the PR to ensure the change does what is expected. 
+    1.  [ ] Ensure UI changes look good without unexpected behavior.  
+
+1. Design  
+    1.  [ ] Do the interactions of the various pieces of code in the PR make sense? 
+    1.  [ ] Does the code recognize and incorporate architectures and coding patterns? 
+
+## Code Quality Pass
+
+1. Complexity
```
Comment (TessFerrandez) [inline code-reviews/readme.md]: Here we have multiple 1.s for example -- linting will catch this

File: code-reviews/pull-request-template.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+### Link to the design document that prompted this PR?
+- Provide a link to the design document relevant to this area. 
+
+### What are you trying to address ?
+-  Describe the current behavior that you are modifying, and link to issue number
+
+### Description of new changes
+- Write a detailed description of all changes
+- If you have only done mechanical (not logical) changes point that out here
+-  Describe the new behavior that is introduces
```
[the remark this answers is not in the document]
Comment (fnocera) [reply code-reviews/pull-request-template.md]: Agree with "that is introduced"

File: code-reviews/readme.md [hunk 6]
```diff
@@ -1,140 +1,160 @@
...
-### Q: What if we do pair programming?
-
-A: Someone outside of the pair should perform the code review. One of the other major benefits of code reviews is spreading knowledge about the code base to other members of the team that don't usually work in the part of the codebase under review.
-
-### Q: What if we do mob programming?
-
-A: A member of the mob who spent less (or no) time at the keyboard should perform the code review.
+# Code Reviews
+
+Developers working on [CSE](../../CSE.md) projects should conduct peer code reviews on every pull request (or check-in to a shared branch).
```
Comment (TessFerrandez) [inline code-reviews/readme.md]: Link is no longer valid

File: code-reviews/readme.md [hunk 7]
```diff
@@ -1,140 +1,160 @@
...
+1. Tests 
+    1.  [ ] Tests should always be committed in the same PR as the code itself (‘I’ll add tests next’ is not acceptable) 
+    1.  [ ] Make sure tests are sensible and valid assumptions are made. 
+
+## Language Specific Guidance
+
+-   [Bash](recipes/Bash.md)
+-   [C#](recipes/CSharp.md)
+-   [Go](recipes/Go.md)
+-   [Java](recipes/Java.md)
```
Comment (TessFerrandez) [inline code-reviews/readme.md]: Java.md does not exist

File: code-reviews/pull-request-template.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (debugerr) [reply code-reviews/pull-request-template.md]: nit: Probably looking a the latest iteration, hence bringing it up again, there is, I think a typo "introduces" -> "introduced"

File: code-reviews/readme.md (hunk 7 above)
Comment (saroup) [reply code-reviews/readme.md]: good catch!

---
In reply to: [379449240](https://github.com/microsoft/code-with-engineering-playbook/pull/148#discussion_r379449240) [](ancestors = 379449240)

**Comment by saroup** [comment]
thanks Tess for the suggestions! I have actually never used a linter for markdown :)

---
In reply to: [358917834](https://github.com/microsoft/code-with-engineering-playbook/pull/148#issuecomment-358917834) [](ancestors = 358917834)

**Comment by saroup** [comment]
good catch! thank you 

---
In reply to: [358955611](https://github.com/microsoft/code-with-engineering-playbook/pull/148#issuecomment-358955611) [](ancestors = 358955611)

File: code-reviews/pull-request-template.md (hunk 1 above)
Comment (saroup) [reply code-reviews/pull-request-template.md]: ran the markdown linter 

---
In reply to: [379418168](https://github.com/microsoft/code-with-engineering-playbook/pull/148#discussion_r379418168) [](ancestors = 379418168)

**Comment by TessFerrandez** [comment]
Looks great.  Thanks for the work.
I am approving, just a small comment -- as part of fixing the markdown style I noticed you removed the [ ] checkbox style items - they could have been fixed with * [ ] in case you wanted to keep them.  

I will leave to you to decide what you thinks looks best and more consistent with the rest of the docs.

**Pull request merged**